wire-moshi-support/src/main/java/com/squareup/wire/ByteStringJsonAdapter.java

```diff
@@ -0,0 +1,31 @@
+package com.squareup.wire;
+
+import com.squareup.moshi.JsonAdapter;
+import com.squareup.moshi.JsonReader;
+import com.squareup.moshi.JsonWriter;
+import java.io.IOException;
+import javax.annotation.Nullable;
+import okio.ByteString;
+
+import static com.squareup.moshi.JsonReader.Token;
+
+class ByteStringJsonAdapter extends JsonAdapter<ByteString> {
+  @Override
+  public void toJson(JsonWriter out, @Nullable ByteString byteString) throws IOException {
+    if (byteString == null) {
+      out.nullValue();
+    } else {
+      out.value(byteString.base64());
+    }
+  }
+
+  @Nullable
```

Comment:
nit: this is a type annotation so it should immediately precede the type it applies to, `ByteString`